.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,235 @@
+
+@Library('apm@current') _
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-agent-java'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    DOCKERHUB_SECRET = 'secret/apm-team/ci/elastic-observability-dockerhub'
+    ELASTIC_DOCKER_SECRET = 'secret/apm-team/ci/docker-registry/prod'
+    NEXUS_SECRET = 'secret/apm-team/ci/nexus'
+    MAVEN_CONFIG = '-Dmaven.repo.local=.m2'
+    OPBEANS_REPO = 'opbeans-java'
+
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+
+  }
+  parameters {
+    string(name: 'MAVEN_CONFIG', defaultValue: '-B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn', description: 'Additional maven options.')
+    string(name: 'branch_specifier', defaultValue: 'master')
+  }
+
+  stages {
+    stage('Initializing'){
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        JAVA_HOME = "${env.HUDSON_HOME}/.java/java10"
+        PATH = "${env.JAVA_HOME}/bin:${env.PATH}"
+        MAVEN_CONFIG = "${params.MAVEN_CONFIG} ${env.MAVEN_CONFIG}"
+      }
+      stages {
+        /**
+         Checkout the code
+        */
+        stage('Checkout') {
+          steps {
+            dir("${BASE_DIR}") {
+              // This is required to ensure that Maven is working with a real branch
+              sh(script: "git checkout -f {$branch_specifier}")
+            }
+          }
+        }
+      }
+    }
+    stage('Release') {
+      options { skipDefaultCheckout () }
+      stages{
+        // 1. Check to see if oss.sonatype.org is up. We throw an exception and fail the job if there is no response.
+        stage('Check oss.sonatype.org') {
+          steps {
+            // If this fails, an exception should be thrown and execution will halt
+            httpRequest(url: "https://oss.sonatype.org/#welcome")
+          }
+        }
+        // 2. Review project version.
+        stage('Review project version') {
+          steps {
+            script {
+              // Show the select input modal
+              def INPUT_PARAMS = input message: 'Do you wish to update the version?', ok: 'Next', parameters: [choice(name: 'UPDATE_CHOICE', choices:['Yes', 'No'], description: "This will run `mvn release:update-versions`")]
+              if (INPUT_PARAMS == 'Yes') {
+                  dir("${BASE_DIR}") {
+                    sh mvn release:update-versions
+                  }
+              } else {
+                  echo 'Skipping version update'
+              }
+            }
+          }
+        }
+        // 3. Execute the release Jenkins job on the internal ci server.
+            // For now we assume that the release has been pushed to staging already on the internal CI
```

Comment:
How does the release job get triggered?

Reply:
By hand. I could probably improve this description but what I meant to indicate here is that this pipeline should be run after the existing release job is run.

Reply:
maybe open a dialog describing what to do which should be closed after the release job is done?

Reply:
BTW, I love that this just delegates to the existing and battle-tested release job instead of re-inventing this step which is definitely the most critical piece.

Reply:
Or we could trigger the release job when a new tag is created. That way, we don't have a direct dependency to actively trigger the build in the internal CI form the public CI.

The public CI  would execute the [`release:prepare`](http://maven.apache.org/maven-release/maven-release-plugin/examples/prepare-release.html) phase of the maven release plugin which updates the versions and creates a new tag.

The creation of the tag triggers the release job on the internal CI which executes the [`release:perform`](http://maven.apache.org/maven-release/maven-release-plugin/examples/perform-release.html) phase of the release plugin.

This does not have to, and probably shouldn't, be in the first iteration of the automated release.

Reply:
I agree that this is a good long-term goal and I also agree that we shouldn't try to achieve this in the first iteration. Therefore, I'm going to resolve this comment because it's becoming a bit tricky to track all the feedback in this PR but we'll come back to it after we have this PR merged. Thanks, @felixbarny ! :)